php/WP_Mock/Exceptions/WP_MockException.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace WP_Mock\Exceptions;
+
+use Exception;
+use Throwable;
+
+/**
+ * Base WP_Mock Exception.
+ */
+class WP_MockException extends Exception
+{
+    /** @var int default from {@see Exception} - child implementations of this class may override this */
+    protected $code = 0;
+
+    /**
+     * Exception constructor.
+     *
+     * @param string $message
+     * @param Throwable|null $previous
+     */
+    public function __construct(string $message = '', Throwable $previous = null)
```

Comment:
Why not include the code in the constructor? (I did see your comment about child implementations allowing for overrides, but just curious why not put it in the constructor anyway.)

Reply:
I checked our usage in MWC packages we also seem to put the code directly in the exception property - we also tend to use codes mapped to HTTP errors sometimes, which I doubt is going to happen in WP_Mock.